### ISSUE TITLE
Expand medical API for third parties

### ISSUE THREAD

**Issue by thojkooi** [acemod/ACE3#4387]
Expand medical API for third parties to support integration and status detection.

Example:

> API for disabling ACE Medical screen effects
> This would be very useful when starting a spectator script while unconscious for training purposes (eg: see how your team handles your body in a mass casualty incident).
> 
> I can't fight the main medical loop here:  https://github.com/acemod/ACE3/blob/b489750d5b1276753c65f46c9e2c6062b61cc6fd/addons/medical/XEH_postInit.sqf#L108 
> and it would force me to give unconscious players a neutered zeus.
> 
> It could be as simple as having line 112 in /addons/medical/XEH_postInit.sqf also check if a global variable that is exposed to mission makers is true/false, in addition to checking if the player is using zeus or is dead. I'm not sure if this is in line with ACE coding standards though.
> By @adampiskorski

Will pick up as part of the medical rewrite (but only after the other functionality is done. This is a nice to have).

**Comment by jonpas** [comment]
Does that include gentle and setter of the entire medical status? That would be great for any persistence scenarios.

**Comment by adampiskorski** [comment]
A single array that fully describes the medical status of a unit that can be stored would be great. We use a script to restore states mid mission in the event of crashes or disconnects and this would be great. ATM i just set damage

**Comment by Insnee** [comment]
Yes, getter & setter to store and load the medical status would be nice!

**Comment by commy2** [comment]
`MedicInfo = player call ace_medical_fnc_serializeUnit`
-> MedicInfo is array of numbers containing pain, wounds, bloodloss, unconscioussness

`[player, MedicInfo] call ace_medical_fnc_deserializeUnit`

**Comment by jonpas** [comment]
`ace_medical_fnc_getUnitStatus` and `ace_medical_fnc_setUnitStatus` might be clearer.

**Comment by bux578** [comment]
I agree with @jonpas 

Maybe even:
`ace_medical_fnc_api_getUnitStatus`

Or it could be a separate pbo:
`ace_medical_api_fnc_getUnitStatus`

Or it could be even less specific:
`ace_api_fnc_getUnitMedicalStatus`

**Comment by Arcanum417** [comment]
@adampiskorski I am sorry to go off topic here, but I would be quite interested in that script, right now I am using really simple version of this, it only saves objects and empty vehicles.
To not clutter this issue more please write me on discord Arcanum#3037

**Comment by KHunter-Arma** [comment]
I don't want to start a big discussion on this old topic and I'm sure you guys have done some great work on the rewrite since the last reply, but I just want to say my persistency mod already works with the current release (non-rewrite) version of ACE Medical, and if you still have work that needs to be done for the rewrite regarding persistency support, I'd be happy to contribute. I'd volunteer for more if I had the time...

**Comment by Mike-MF** [comment]
Closing due to age, inactivity or already solved. May be re-opened if the issue is still present, even better just open a new up-to-date Issue.

**Comment by jonpas** [comment]
That has been done, there are now more API functions as well as general get/set.
